### PR TITLE
NNS1-3486: new util function to convert from period to dateRange

### PR DESCRIPTION
**Pull request by yhabib**
# Motivation

We want to convert the selected radio value into a `DateRange` object that can be used when fetching the translations.

# Changes

- New utility `convertPeriodToNanosecondRange` to convert a reporting period into a transaction date range.

# Tests

- Unit tests

# Todos

- [ ] Add entry to changelog (if necessary).
Not necessary

### CODE REVIEW

File: frontend/src/lib/utils/reporting.utils.ts [hunk 1]
```diff
@@ -459,3 +463,27 @@ export const buildNeuronsDatasets = ({
...
+    };
+  }
+
+  if (period === "year-to-date") {
+    return {
+      from: BigInt(new Date(currentYear, 0, 1).getTime()),
+    };
+  }
+
+  throw new Error(`Invalid period: ${period}`);
```
[the remark this answers is not in the document]
Comment (yhabib) [reply frontend/src/lib/utils/reporting.utils.ts]: `switch` can do the trick.

File: frontend/src/lib/utils/reporting.utils.ts [hunk 2]
```diff
@@ -459,3 +463,27 @@ export const buildNeuronsDatasets = ({
...
+): TransactionsDateRange => {
+  if (period === "all") return {};
+
+  const now = new Date();
+  const currentYear = now.getFullYear();
+
+  if (period === "last-year") {
+    return {
+      from: BigInt(new Date(currentYear - 1, 0, 1).getTime()),
+      to: BigInt(new Date(currentYear - 1, 11, 31, 23, 59, 59, 999).getTime()),
```
[the remark this answers is not in the document]
Comment (yhabib) [reply frontend/src/lib/utils/reporting.utils.ts]: I'll review the exclusion nature of the service in a follow-up.

File: frontend/src/lib/utils/reporting.utils.ts [hunk 3]
```diff
@@ -1,8 +1,8 @@
+import type { TransactionResults } from "$lib/services/reporting.services";
```
Comment (dskloetd) [inline frontend/src/lib/utils/reporting.utils.ts:1]: `TransactionResults` should be moved to `$lib/types` as well in another PR.

File: frontend/src/lib/types/reporting.ts [hunk 4]
```diff
@@ -1 +1,6 @@
 export type ReportingDateRange = "all" | "last-year" | "year-to-date";
+
+export type TransactionsDateRange = {
+  from?: bigint;
+  to?: bigint;
```
Comment (dskloetd) [inline frontend/src/lib/types/reporting.ts:5]: Add a comment about whether `to` is inclusive or exclusive.

**Review by dskloetd — APPROVED** [review]

**Pull request merged**